jmeter/src/main/java/org/eclipse/hono/jmeter/HonoCommanderSampler.java

```diff
@@ -0,0 +1,207 @@
+import org.slf4j.LoggerFactory;
+
+/**
+ * JMeter creates an instance of a sampler class for every occurrence of the element in every thread. [some additional
+ * copies may be created before the test run starts]
+ *
+ * Thus each sampler is guaranteed to be called by a single thread - there is no need to synchronize access to instance
+ * variables.
+ *
+ * However, access to class fields must be synchronized.
+ */
+public class HonoCommanderSampler extends HonoSampler implements ThreadListener {
+
+    private static final org.slf4j.Logger LOG = LoggerFactory.getLogger(HonoCommanderSampler.class);
+    private static final String USE_SENDER_TIME = "useSenderTime";
+    private static final String SENDER_TIME_IN_PAYLOAD = "senderTimeInPayload";
+    private static final String SENDER_TIME_VARIABLE_NAME = "senderTimeVariableName";
+    private static final String RECONNECT_ATTEMPTS = "reconnectAttempts";
+    private static final String DEFAULT_SENDER_TIME_VARIABLE_NAME = "timeStamp";
+    private static final String COMMAND = "command";
+    private static final String COMMAND_PAYLOAD = "commandPayload";
+    private static final String COMMAND_TIMEOUT = "commandTimeOut";
+    private static final String PROTOCOL = "protocol";
+    private static final int DEFAULT_COMMAND_REQUEST_TIMEOUT = 1000;
+    private HonoCommander honoCommander;
+
+    @Override
+    public SampleResult sample(final Entry entry) {
+        final HonoCommanderSampleResult sampleResult = new HonoCommanderSampleResult();
+        sampleResult.setDataType(SampleResult.TEXT);
+        sampleResult.setSampleLabel(getName());
+        honoCommander.sample(sampleResult);
+        return sampleResult;
+    }
+
+    @Override
+    public void threadStarted() {
+        LOG.info("Sampler thread started");
+        try {
+            honoCommander = new HonoCommander(this);
+            honoCommander.start().join();
+        } catch (final CompletionException e) {
+            throw new RuntimeException(format("Error starting commander sampler for tenant %s", getTenant()),
+                    e.getCause());
+        }
+    }
+
+    @Override
+    public void threadFinished() {
+        LOG.info("Sampler thread finished");
+        if (honoCommander != null) {
+            try {
+                honoCommander.close().join();
+            } catch (final CompletionException e) {
+                LOG.error("error during shut down of command application", e);
+            }
+        }
+
+    }
+
+    public boolean isUseSenderTime() {
+        return getPropertyAsBoolean(USE_SENDER_TIME);
+    }
+
+    /**
+     * Sets whether or not to use the sender time.
+     *
+     * @param useSenderTime {@code true} in order to use the sender time.
+     */
+    public void setUseSenderTime(final boolean useSenderTime) {
+        setProperty(USE_SENDER_TIME, useSenderTime);
+    }
+
+    @Override
+    public String getEndpoint() {
```

Comment:
Why is this method overridden? It doesn't seem to add anything? Also see `setEndpoint`.

Reply:
Thank you for bringing it to notice. I will remove them.